runtime/src/main/scala/com/ing/baker/runtime/java_api/JBaker.scala

```diff
@@ -48,11 +50,260 @@ class JBaker(actorSystem: ActorSystem, implementations: java.lang.Iterable[AnyRe
+    * @param processId The process identifier
+    * @return
+    */
+  @throws[NoSuchProcessException]("When no process exists for the given id")
+  @throws[TimeoutException]("When the process does not respond within the default deadline")
+  def getIngredients(processId: String): java.util.Map[String, Value] =
+    getIngredients(processId, defaultTimeout)
+
+  /**
+    * Returns all the ingredients that are accumulated for a given process.
+    *
+    * @param processId The process identifier
+    * @return
+    */
+  @throws[NoSuchProcessException]("When no process exists for the given id")
+  @throws[TimeoutException]("When the process does not respond within the default deadline")
+  def getIngredients(processId: UUID): java.util.Map[String, Value] =
+    getIngredients(processId.toString)
+
+  /**
+    * Returns all events that have occurred for a given process.
+    *
+    * Note that this list is eventually consistent. This means that it might take some
+    * time before an event that occurred in the process is appended to the list.
+    *
+    * @param processId         The process identifier
+    * @param waitTimeoutMillis The maximum wait time
+    * @return
+    *
+    */
+  def getEvents(processId: String, waitTimeoutMillis: Long): EventList =
+    new EventList(baker.events(processId)(waitTimeoutMillis milliseconds))
+
+  /**
+    * Returns all events that have occurred for a given process.
+    *
+    * Note that this list is eventually consistent. This means that it might take some
+    * time before an event that occurred in the process is appended to the list.
+    *
+    * @param processId         The process identifier
+    * @param waitTimeoutMillis The maximum wait time
+    * @return
+    *
+    */
+  def getEvents(processId: UUID, waitTimeoutMillis: Long): EventList = getEvents(processId.toString, waitTimeoutMillis)
+
+  /**
+    * Returns all events that have occurred for a given process.
+    *
+    *
+    * Note that this list is eventually consistent. This means that it might take some
+    * time before an event that occurred in the process is appended to the list.
+    *
+    * @param processId The process identifier
+    * @return
+    */
+  def getEvents(processId: String): EventList = getEvents(processId, defaultTimeout)
+
   /**
-    * Return the handler for a specific recipe
+    * Returns all events that have occurred for a given process.
+    *
+    * Note that this list is eventually consistent. This means that it might take some
+    * time before an event that occurred in the process is appended to the list.
     *
-    * @param name
+    * @param processId The process identifier
     * @return
     */
-  def getRecipeHandler(name: String): JRecipeHandler = new JRecipeHandler(baker.getRecipeHandler(name))(actorSystem)
+  def getEvents(processId: UUID): EventList = getEvents(processId.toString, defaultTimeout)
+
+  /**
+    * Returns the compiled recipe.
+    *
+    * @return The compiled recipe
+    */
```

Comment:
Could throw a TimeoutException, could add something like: 

```@throws[TimeoutException]("When baker does not respond within the default deadline")```

Same for some other methods.

Reply:
Will do this in the other branch where we already refactor the timeout part for Baker/JBaker.